sdk/core/azure-core/inc/azure/core/azure_assert.hpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Provide assert macros to use with pre-conditions.
+ *
+ * @remark Asserts are turned ON when `NDEBUG` is NOT defined (for Debug build). For Release build,
+ * `std::abort()` is directly called if the condition is false, whitout calling assert().
+ *
+ */
+
+#pragma once
+
+#if defined(NDEBUG)
+
+/*
+ * NDEBUG = defined = Build is on Release
+ * Define AZURE_ASSERT to call abort directly on exp == false
+ */
+
+#include <cstdlib>
+
+#define AZURE_ASSERT(exp) \
+  if (!(exp)) \
+  { \
+    std::abort(); \
+  }
+
+#define AZURE_ASSERT_MSG(exp, msg) AZURE_ASSERT(exp)
+
+#else
+
+/*
+ * NDEBUG = NOT defined = Build is on Debug
+ * Define AZURE_ASSERT to call assert to provide better debug experience.
+ */
+
+#include <cassert>
+
+#define AZURE_ASSERT(exp) assert((exp))
```

Comment:
I suggest we abort() if the user clicks continue.  Continuing past an assert on debug is problematic.

Reply:
Clarifying: The implementation of `assert` in the Windows UCRT (and the older CRT that predates UCRT) is non-conforming in that it can return when a user specifically asks to do so in the debugger.